tests/e2e_test.go

```diff
@@ -63,11 +73,31 @@ var _ = ginkgo.BeforeSuite(func() {
 		teleporterContractAddress,
 		fundedKey,
 	)
+
+	var ctx context.Context
+	ctx, cancelDecider = context.WithCancel(context.Background())
+	// we'll call cancelDecider in AfterSuite, but also call it if this
+	// process is killed, because AfterSuite won't always run then:
+	signalChan := make(chan os.Signal, 2)
+	signal.Notify(signalChan, os.Interrupt, syscall.SIGTERM)
+	go func() {
+		<-signalChan
+		cancelDecider()
+	}()
+	decider = exec.CommandContext(ctx, "./tests/cmd/decider/decider")
+	decider.Start()
+	log.Info("Started decider service")
+
 	log.Info("Set up ginkgo before suite")
 })
 
 var _ = ginkgo.AfterSuite(func() {
-	localNetworkInstance.TearDownNetwork()
+	if localNetworkInstance != nil {
+		localNetworkInstance.TearDownNetwork()
+	}
+	if decider != nil {
+		cancelDecider()
+	}
```

Comment:
We don't actually need this change either, afaik. Cleanup should only need to remove any files written that we want removed.

Reply:
I took this part out and then the decider process was still running after the e2e tests completed successfully. I think we need it.